torch/jit/__init__.py

```diff
@@ -580,5 +567,85 @@ def _time(trace_name, name, time=True):
         print("{} {} time: {} ms".format(trace_name, name, start.elapsed_time(end)))
 
 
+def verify(model, args, loss_fn=torch.sum, devices=None):
+    """
+    Verify that a JIT compiled model has the same behavior as its uncompiled
+    version along with its backwards pass.
+
+    This function has side-effects (e.g., it executes your model / saves and loads
+    parameters), so don't expect the model to come out exactly the same as what
+    you passed in.
```

Comment:
Executing a model and saving/loading parameters isn't an externally visible side effect. Let's just say that it **might have side effects if your model changes at every forward (e.g. you have BN in training mode)**. It might also be good switch the model to eval like we do with exports

Reply:
So, the saving/loading parameters is done *solely* so that we can rerun the model with the compiled trace, but otherwise same initial conditions; we're not actually resetting the model to the very original parameter setting. We could do this (so that it has "no side effects") but there is more work to do; for example, you would have to save and restore the state of all the grads, and it just seems like complication for no good reason.

One thing to note is switching model to `eval` does NOT make sense, because a user may very well be interested in knowing that the model runs in training mode the same as compiled and as not.

Reply:
I'm confused. Is it saving as in `torch.save` or is it just clone? How is that a visible side effect? It's only executing the model that can mutate things

Reply:
BTW the default `loss_fn` will break as soon as someone has more than one output of the model. Might be worth fixing or at least mention that in the docs